### PR TITLE
workbench: Adds actions for moveEditorToFirstGroup, moveEditorToSecondGroup, moveEditorToThirdGroup

### LINKED ISSUE #40525: Allow to move editor to specific group
- VSCode Version: 1.19.0
- OS Version: Ubuntu 17.10

I'm coming from Sublime where I used extensively editor groups, and there were keybindings for moving an editor to specific editor group (e.g. <kbd>ctrl</kbd>+<kbd>shift</kbd>+<kbd>1</kbd> moves an editor to the first group, <kbd>ctrl</kbd>+<kbd>shift</kbd>+<kbd>2</kbd> to the second etc.). Here in Code we have commands like `moveEditorToNextGroup` and `moveEditorToPreviousGroup`, along with `focusFirstEditorGroup`, `focusSecondEditorGroup` etc.

It would be handy if we could move an editor to specific group with shortcut, not only to the left/to the right with commands like `workbench.action.moveEditorToFirstGroup`, `workbench.action.moveEditorToSecondGroup` etc.

### PR DESCRIPTION
**Pull request by shobhitchittora**
Adds the following actions for workbench tabs - 

(For Mac Only - TODO: Need to confirm the same in Win & UNIX)

1. moveEditorToFirstGroup -  `^ ⌘ 1`
2. moveEditorToSecondGroup - `^ ⌘ 2`
3. moveEditorToThirdGroup - `^ ⌘ 3`

Fixes: #40525

### CODE REVIEW

**Comment by shobhitchittora** [comment]
@bpasero Are there any test to extend upon?

File: src/vs/workbench/browser/parts/editor/editor.contribution.ts [hunk 1]
```diff
@@ -363,6 +363,9 @@ registry.registerWorkbenchAction(new SyncActionDescriptor(MoveGroupLeftAction, M
...
 registry.registerWorkbenchAction(new SyncActionDescriptor(MoveEditorToPreviousGroupAction, MoveEditorToPreviousGroupAction.ID, MoveEditorToPreviousGroupAction.LABEL, { primary: KeyMod.CtrlCmd | KeyMod.Alt | KeyCode.LeftArrow, mac: { primary: KeyMod.CtrlCmd | KeyMod.WinCtrl | KeyCode.LeftArrow } }), 'View: Move Editor into Previous Group', category);
 registry.registerWorkbenchAction(new SyncActionDescriptor(MoveEditorToNextGroupAction, MoveEditorToNextGroupAction.ID, MoveEditorToNextGroupAction.LABEL, { primary: KeyMod.CtrlCmd | KeyMod.Alt | KeyCode.RightArrow, mac: { primary: KeyMod.CtrlCmd | KeyMod.WinCtrl | KeyCode.RightArrow } }), 'View: Move Editor into Next Group', category);
+registry.registerWorkbenchAction(new SyncActionDescriptor(MoveEditorToFirstGroupAction, MoveEditorToFirstGroupAction.ID, MoveEditorToFirstGroupAction.LABEL, { primary: KeyMod.CtrlCmd | KeyMod.Alt | KeyCode.KEY_1, mac: { primary: KeyMod.CtrlCmd | KeyMod.WinCtrl | KeyCode.KEY_1 } }), 'View: Move Editor into First Group', category);
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/editor/editor.contribution.ts]: @shobhitchittora keybinding `KeyMod.CtrlCmd | KeyMod.Alt | KeyCode.KEY_1` is already taken by "Toggle Editor Layout" and cannot be used again

File: src/vs/workbench/browser/parts/editor/editorActions.ts [hunk 2]
```diff
@@ -1553,3 +1553,78 @@ export class MoveEditorToNextGroupAction extends Action {
 		return TPromise.as(true);
 	}
 }
+
+export class MoveEditorToFirstGroupAction extends Action {
+
+	public static readonly ID = 'workbench.action.moveEditorToFirstGroupAction';
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/editor/editorActions.ts]: @shobhitchittora remove the trailing `Action` from the action ids (e.g. `workbench.action.moveEditorToFirstGroup`)

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/browser/parts/editor/editor.contribution.ts (hunk 1 above)
Comment (shobhitchittora) [reply src/vs/workbench/browser/parts/editor/editor.contribution.ts]: Correct me if I'm wrong - 

1. The primary key binding scheme is applied to Windows and Linux builds. Mac has it's own primary binding object specified.
2. `KeyMod.CtrlCmd | KeyMod.Alt | KeyCode.KEY_1` is MAC-primary for "Toggle Editor Layout", while it's WIN/LINUX-primary for "MoveEditorToFirstGroupAction".
3. Any other suggestions?

File: src/vs/workbench/browser/parts/editor/editorActions.ts (hunk 2 above)
Comment (shobhitchittora) [reply src/vs/workbench/browser/parts/editor/editorActions.ts]: Done.

File: src/vs/workbench/browser/parts/editor/editor.contribution.ts (hunk 1 above)
Comment (bpasero) [reply src/vs/workbench/browser/parts/editor/editor.contribution.ts]: @shobhitchittora you are right, sorry I did not see the mac specific keybinding

File: src/vs/workbench/browser/parts/editor/editorActions.ts [hunk 3]
```diff
@@ -1553,3 +1553,75 @@ export class MoveEditorToNextGroupAction extends Action {
 		return TPromise.as(true);
 	}
 }
+
+export class MoveEditorToFirstGroupAction extends Action {
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/editor/editorActions.ts]: @shobhitchittora all these actions are very similar in their implementation. Can we have a base class (abstract) that all classes extend from and in the base class we simply have a method to ask for the position?

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/browser/parts/editor/editor.contribution.ts (hunk 1 above)
Comment (shobhitchittora) [reply src/vs/workbench/browser/parts/editor/editor.contribution.ts]: Thanks for looking into it 👍🏻.

File: src/vs/workbench/browser/parts/editor/editorActions.ts (hunk 3 above)
Comment (shobhitchittora) [reply src/vs/workbench/browser/parts/editor/editorActions.ts]: Yes on it.

File: src/vs/workbench/browser/parts/editor/editor.contribution.ts (hunk 1 above)
Comment (shobhitchittora) [reply src/vs/workbench/browser/parts/editor/editor.contribution.ts]: Done.

**Comment by shobhitchittora** [comment]
@bpasero Any idea why Travis is failing?

File: src/vs/workbench/browser/parts/editor/editorActions.ts (hunk 3 above)
Comment (shobhitchittora) [reply src/vs/workbench/browser/parts/editor/editorActions.ts]: Done 💅🏻.
@bpasero Please review now.

**Comment by bpasero** [comment]
LGTM thanks 👍

**Comment by hardianlawi** [comment]
Hi,

Why couldn't I find these shortcuts in my `keyboard references` now? I think I used to be able to press `shift+Alt+2` and `shift+Alt+3` to move the editors (it will create it if the group does not exist)

Thank you

**Comment by hardianlawi** [comment]
The VSCode version im using

image

**Comment by bpasero** [comment]
@hardianlawi they got removed. Read [this](https://github.com/Microsoft/vscode-docs/blob/vnext/release-notes/v1_25.md#grid-editor-layout) and [this](https://github.com/Microsoft/vscode-docs/blob/vnext/release-notes/v1_25.md#removed-commands).

**Pull request merged**